Makefile

```
@@ -62,19 +70,24 @@ libquirc.so.$(LIB_VERSION): $(LIB_OBJ)
 .c.o:
 	$(CC) $(QUIRC_CFLAGS) -o $@ -c $<
 
+.cxx.o:
+	$(CXX) $(QUIRC_CXXFLAGS) -o $@ -c $<
```

This would not work for me (GNU Make 4.3):
```console
% make quirc-demo-opencv
cc -Ilib -O3 -Wall -fPIC -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT  -o demo/dthash.o -c demo/dthash.c
cc -Ilib -O3 -Wall -fPIC -I/usr/include/SDL -D_GNU_SOURCE=1 -D_REENTRANT  -o demo/demoutil.o -c demo/demoutil.c
make: *** No rule to make target 'demo/demo_opencv.o', needed by 'quirc-demo-opencv'.  Stop.
% make --version
GNU Make 4.3
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

 I had to add a GNU Make specific recipe:
```suggestion
.cxx.o:
	$(CXX) $(QUIRC_CXXFLAGS) -o $@ -c $<
%.o: %.cxx
	$(CXX) $(QUIRC_CXXFLAGS) -o $@ -c $<
```

---

fixed differently